scripts/redeploy.py

```diff
@@ -39,81 +42,98 @@ def create_symlink(source, linkname):
         else:
             raise e
 
+def req_headers():
+    return {
+        "Authorization": "Bearer %s" % (arg_api_token,),
+    }
+
 @app.route("/", methods=["POST"])
-def on_receive_jenkins_poke():
-    # {
-    #    "name": "VectorWebDevelop",
-    #    "build": {
-    #        "number": 8
-    #    }
-    # }
+def on_receive_buildkite_poke():
+    got_webhook_token = request.headers.get('X-Buildkite-Token')
+    if got_webhook_token != arg_webbook_token:
+        print("Denying request with incorrect webhook token: %s" % (got_webhook_token,))
+        abort(400, "Incorrect webhook token")
+        return
+
+    required_api_prefix = None
+    if arg_buildkit_org is not None:
+        required_api_prefix = 'https://api.buildkite.com/v2/organizations/%s' % (arg_buildkit_org,)
+
     incoming_json = request.get_json()
     if not incoming_json:
         abort(400, "No JSON provided!")
         return
     print("Incoming JSON: %s" % (incoming_json,))
 
-    job_name = incoming_json.get("name")
-    if not isinstance(job_name, basestring):
-        abort(400, "Bad job name: %s" % (job_name,))
+    event = incoming_json.get("event")
+    if event is None:
+        abort(400, "No 'event' specified")
         return
 
-    build_num = incoming_json.get("build", {}).get("number", 0)
-    if not build_num or build_num <= 0 or not isinstance(build_num, int):
-        abort(400, "Missing or bad build number")
+    if event == 'ping':
+        print("Got ping request - responding")
+        return jsonify({'response': 'pong!'})
```

Comment:
I guess the response can be anything here?

Reply:
yep